lwpreflight/azure/caller.go

```diff
@@ -0,0 +1,110 @@
+package azure
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization"
+)
+
+type Caller struct {
+	ObjectID    string
+	DisplayName string
+	PrincipalID string
+	TenantID    string
+	IsAdmin     bool // true if the caller has Owner or Contributor role
+}
+
+func FetchCaller(p *Preflight) error {
+	// Get caller identity
+	token, err := p.cred.GetToken(context.Background(), policy.TokenRequestOptions{
+		Scopes: []string{"https://management.azure.com/.default"},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get token: %v", err)
+	}
+
+	// Parse JWT token to get caller info
+	claims, err := parseJWTClaims(token.Token)
+	if err != nil {
+		return fmt.Errorf("failed to parse token claims: %v", err)
```

Comment:
The err message is duplicated with the ones returned from `parseJWTClaims`

```suggestion
		return err
```